src/ucm/util/reloc.c

```diff
@@ -91,6 +91,23 @@ ucm_reloc_get_entry(ElfW(Addr) base, const ElfW(Phdr) *dphdr, ElfW(Sxword) tag)
     return 0;
 }
 
+static void *
+ucm_reloc_get_pointer(ElfW(Addr) base, const ElfW(Phdr) *dphdr, ElfW(Sxword) tag)
+{
+    uintptr_t entry;
+    entry = ucm_reloc_get_entry(base, dphdr, tag);
+    /*
+     * On some combinations of C library and architecture the entries in the
+     * dynamic table are not relocated. As a heuristic check if the value from
+     * the table looks like a pointer or an offset by checking if it is below
+     * the loaded address of the library.
+    */
+    if ((entry > 0) && (entry < base)) {
```

Comment:
it might be possible to avoid heuristic, tried to detail on corresponding issue: https://github.com/openucx/ucx/issues/9283#issuecomment-1673644180